scheduler/src/cook/mesos/api.clj

```diff
@@ -1144,13 +1152,149 @@
      :exists? (partial retrieve-jobs conn)
      :handle-ok (fn [ctx] (render-jobs-for-response-deprecated conn framework-id ctx))}))
 
+(defn valid-name-filter?
+  "Returns true if the provided name filter is either nil or satisfies the JobNameListFilter schema"
+  [name]
+  (or (nil? name)
+      (nil? (s/check JobNameListFilter name))))
+
+(defn normalize-list-states
+  "Given a set, states, returns a new set that only contains one of the
+  'terminal' states of completed, success, and failed. We take completed
+  to mean both success and failed."
+  [states]
+  (if (contains? states "completed")
+    (set/difference states util/instance-states)
+    (if (set/superset? states util/instance-states)
+      (-> states (set/difference util/instance-states) (conj "completed"))
+      states)))
```

Comment:
This might be easier to read:
```
(def terminal states #{"completed" "failed" "success"})
...
(if (some states terminal-states)
  (-> states (set/difference terminal-states) (conj "completed"))
  states)
````

Reply:
This is the current (and desired) behavior:

```bash
(api/normalize-list-states #{"success" "failed" "completed"})
=> #{"completed"}
(api/normalize-list-states #{"success" "failed"})
=> #{"completed"}
(api/normalize-list-states #{"success"})
=> #{"success"}
(api/normalize-list-states #{"failed"})
=> #{"failed"}
(api/normalize-list-states #{"completed"})
=> #{"completed"}
```

I don't think your suggestion will have the same results when `states` is `#{"success"}`, for example.

Reply:
okay